src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -77,7 +78,23 @@ private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMe
 
                         string challengeData = challenge.ChallengeData;
 
-                        string spn = "HTTP/" + authUri.IdnHost;
+                        // Need to use FQDN normalized host so that CNAME's are traversed.
+                        string spn;
+                        if (authUri.HostNameType == UriHostNameType.IPv6 || authUri.HostNameType == UriHostNameType.IPv4)
+                        {
+                            spn = "HTTP/" + authUri.IdnHost;
```

Comment:
Do we need to place brackets around the IPv6 address to work around #28863?

Reply:
>Do we need to place brackets around the IPv6 address to work around #28863?

I don't think we have tested this scenario.  Both .NET Framework and .NET Core currently have this as-is.  And my PR doesn't change this.

Practically speaking, NT Auth is likely to work in a scenario where IP addresses are used in the URI.  IP address based SPNs are not registered by default in Kerberos (Windows Active Directory). So, they would have to be custom registered by an Enterprise. And typically that would not include any brackets. But I've never seen Enterprises even use IP addresses for SPNs.

The safe thing to do for this PR is to leave the behavior as-is, and not add brackets right now. But we could open up an issue for further investigation if we want.

Reply:
In that case this seems like a reasonable behavior.